agent/src/storage/blobs.rs

```diff
@@ -36,51 +44,20 @@ impl BlobStore {
         Ok(Self { location })
     }
 
-    /// Fetch a manifest by its fully-qualified name.
-    pub async fn manifest(&self, fq_name: &str) -> Result<Manifest, ServalError> {
-        let bytes = cacache::read(&self.location, Manifest::make_manifest_key(fq_name)).await?;
-        if let Ok(data) = String::from_utf8(bytes) {
-            let manifest: Manifest = toml::from_str(&data)?;
-            Ok(manifest)
-        } else {
-            // TODO: bad data error
-            Err(ServalError::ManifestNotFound(fq_name.to_string()))
-        }
-    }
-
-    /// Store a job type manifest. Returns the integrity checksum.
-    pub async fn store_manifest(&self, manifest: &Manifest) -> Result<Integrity, ServalError> {
-        let toml = toml::to_string(manifest)?;
-        let meta_sri = cacache::write(&self.location, manifest.manifest_key(), &toml).await?;
-        Ok(meta_sri)
-    }
-
-    /// Store a job with metadata and an executable for later use. Returns the integrity checksums for the pair.
-    pub async fn store_manifest_and_executable(
-        &self,
-        manifest: &Manifest,
-        executable: &[u8],
-    ) -> Result<(Integrity, Integrity), ServalError> {
-        let toml = toml::to_string(manifest)?;
-        let meta_sri = cacache::write(&self.location, manifest.manifest_key(), &toml).await?;
-        let exec_sri =
-            cacache::write(&self.location, manifest.executable_key(), executable).await?;
-
-        Ok((meta_sri, exec_sri))
-    }
-
-    /// Store an executable in our blob store by its fully-qualified manifest name and a version string.
-    pub async fn store_executable(
+    /// Fetch an executable by key as a read stream.
+    pub async fn executable_as_stream(
         &self,
         name: &str,
         version: &str,
-        bytes: &[u8],
-    ) -> Result<String, ServalError> {
+    ) -> Result<ReaderStream<Reader>, ServalError> {
         let key = Manifest::make_executable_key(name, version);
-        let sri = cacache::write(&self.location, key, bytes).await?;
-        Ok(sri.to_string())
+        let fd = cacache::Reader::open(&self.location, key).await?;
+        let stream = ReaderStream::new(fd);
+        Ok(stream)
     }
 
+    // Unused; consider removing
```

Comment:
I like to put dates in comments like these so the next person to casually read it can have the realization "wow, this has been apparently unused for $long_time" and is more likely to do something about it.